### PR TITLE
Local planner

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

The current greedy follower wasn't designed in a world were sliding was disabled and action noise existed, so it isn't great in those situations.  This PR switches it out for a local motion primitive planner that does a considerably better job and also sets things up for action noise (by allowing the user to specify what key should be emitted for each action to take as the planner needs to plan on noiseless actions).

Compared to the old one, there is a bump of ~0.02 SPL when sliding is enabled.  When sliding is disabled however, SPL is increased from ~0.85 (with a ~12% failure rate) to ~0.95 (with a ~2% failure rate), which is much better.

## How Has This Been Tested

Via the tests!

## Types of changes

-  New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/esp/nav/GreedyFollower.h [hunk 1]
```diff
@@ -8,111 +8,155 @@
...
   /**
-   * Helper typedef for a sixdof pos
-   **/
-  typedef std::tuple<vec3f, quatf> State;
+   * @brief Helper for a sixdof pose
+   */
+  struct SixDofPose {
+    quatf rotation;
+    vec3f translation;
+  };
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply src/esp/nav/GreedyFollower.h:56]: Yeah, that makes sense

**Comment by erikwijmans** [comment]
For thrashing, it occurs when you enter a loop of left->right->left->right->left->right....  the way to get out of that is to take the best primitive and execute it fully instead of executing the first action and re-planning.

For the situation you described (no action better than just colliding), that results in an error (the cpp code returns an ERROR token and python then throws an error).

**Comment by erikwijmans** [comment]
As a side note, I haven't seen a case where this thing actually thrashes.  I am not sure if you actually need that logic since we have access to the ground truth navmesh to plan on, but its simple enough to have it :)

**Comment by erikwijmans** [comment]
@aclegg3 Can you double check esp::core::RigidState

**Comment by aclegg3** [comment]
> @aclegg3 Can you double check esp::core::RigidState

Looks good. Thanks!

**Comment by mathfac** [comment]
The current greedy follower wasn't designed in a world were sliding was disabled and action noise existed, so it isn't great in those situations. This PR switches it out for a local motion primitive planner that does a considerably better job and also sets things up for action noise (by allowing the user to specify what key should be emitted for each action to take as the planner needs to plan on noiseless actions).

Compared to the old one, there is a bump of ~0.02 SPL when sliding is enabled. When sliding is disabled however, SPL is increased from ~0.85 (with a ~12% failure rate) to ~0.95 (with a ~2% failure rate), which is much better.

**Pull request merged**